portal_spa/src/components/custom/SelectAttributes.tsx

```diff
@@ -1,163 +1,231 @@
+import { useMemo, useState } from "react";
 import {
-  FormField,
-  FormItem,
-  FormLabel,
-  FormControl,
-  FormMessage,
-} from "../ui/form";
-import {
-  Select,
-  SelectTrigger,
-  SelectValue,
-  SelectContent,
-  SelectItem,
-} from "../ui/select";
-import { Input } from "../ui/input";
-import type { Credential } from "@/models/credential";
-import { useFormContext } from "react-hook-form";
-import {
-  Tooltip,
-  TooltipTrigger,
-  TooltipContent,
-} from "@/components/ui/tooltip";
-import { Info } from "lucide-react";
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Checkbox } from "@/components/ui/checkbox";
+import { Textarea } from "@/components/ui/textarea";
+import { cn } from "@/lib/utils";
+import type { Credential, CredentialAttribute } from "@/models/credential";
+import { Check, ChevronDown } from "lucide-react";
+import { Select, SelectContent, SelectItem, SelectTrigger } from "../ui/select";
+import { SelectValue } from "@radix-ui/react-select";
 
-interface Props {
+interface CredentialAttributeFieldProps {
   index: number;
-  credentials: Credential[];
-  onRemove: () => void;
+  credentials: Array<Credential>;
+  value: CredentialAttribute;
+  onChange: (updated: CredentialAttribute) => void;
 }
 
 export default function CredentialAttributeFields({
-  index,
   credentials,
-}: Props) {
-  const form = useFormContext();
-  const { control } = form;
+  value,
+  onChange,
+}: CredentialAttributeFieldProps) {
+  const [open, setOpen] = useState(false);
+  const [search, setSearch] = useState("");
+  const [envFilter, setEnvFilter] = useState({
+    production: true,
+    staging: false,
+    demo: false,
+  });
+
+  const toggleEnv = (env: keyof typeof envFilter) => {
+    setEnvFilter((prev) => ({ ...prev, [env]: !prev[env] }));
+  };
+
+  const filteredCredentials = useMemo(() => {
+    return credentials.filter((cred) => {
+      const matchesEnv = envFilter[cred.environment as keyof typeof envFilter];
+
+      const inCredentialName =
+        cred.name_en.toLowerCase().includes(search.toLowerCase()) ||
+        cred.name_nl.toLowerCase().includes(search.toLowerCase());
+
+      const inAttributeName = cred.attributes?.some((attr) =>
+        [attr.name_en, attr.name_nl]
+          .filter(Boolean)
+          .some((name) => name?.toLowerCase().includes(search.toLowerCase()))
+      );
+
+      return matchesEnv && (inCredentialName || inAttributeName);
+    });
+  }, [credentials, envFilter, search]);
+
+  const selected = credentials.find((cred) => cred.id === value?.credential_id);
+
+  const handleFieldChange = (fieldKey: string, newValue: unknown) => {
+    onChange?.({
```

Comment:
The type definition of `onChange` is not marked as optional. Maybe the optional chaining operator (`?.`) can be omitted or the type definition should be made optional.